make/langtools/tools/propertiesparser/resources/templates.properties

```diff
@@ -84,3 +88,10 @@ suppress.warnings=\
 lint.category=\
   LintCategory.get({0}).get()
 
+diagnostic.flags=\n\
+  '  'Stream.of({0})\n\
+  '    '.map(s -> s.replace(''-'', ''_''))\n\
+  '    '.map(s -> s.toUpperCase(Locale.ROOT))\n\
+  '    '.map(DiagnosticFlag::valueOf)\n\
+  '    '.toArray(DiagnosticFlag[]::new)
```

Comment:
If instead of an array we returned a list, or a set, maybe we could use empty list/set if there's no flag (instead of `null` as we do now?)